### DISCUSSION TITLE
No `pageContext` loaded on new page

### DISCUSSION THREAD

**Discussion by rtritto** [vikejs/vike#855] [Discussion] [answered]
### Description

The title is correct only on first render and it doesn't change when navigate on new page.

### Steps to reproduce:
- clone https://github.com/brillout/vite-plugin-ssr/tree/main/examples/solid-ssr-v1
- create `/pages/index/+config.ts`:
```ts
export default {
  title: 'Home'
}
```
- create `/pages/about/+config.ts`:
```ts
export default {
  title: 'About'
}
```
- add to `/renderer/+config.ts`
```ts
{
  meta: {
    title: {
      env: 'server-and-client'
    }
  },
  passToClient: ['pageProps', 'documentProps', 'title'],
  clientRouting: true
}
```
- change in `/renderer/+onRenderHtml.tsx` (with v1 title is in _config_ and not in _documentProps_)
```diff
- const { documentProps } = pageContext
- const title = (documentProps && documentProps.title) || 'Vite SSR app'
+ const { documentProps, config } = pageContext
+ const title = (config && config.title) || 'Vite SSR app' 
```
- start with _dev_ script
- go to Home page (`/`) --> the title page is Home
- go to About page (`/about`) --> the title page is Home instead of About

Repository: https://github.com/rtritto/vps-no-title

**Comment by brillout** [comment, marked as answer]
See https://github.com/magne4000/solide/blob/1935ce7a93f18ee8c0963938ac21f92f6e7c0dc5/solide/renderer/%2BonRenderClient.tsx#L34-L37. If you want: PR welcome to improve `/examples/solid-ssr-v1/`. (We can leave `/examples/solid-ssr/` as it is: it will soon be removed.)

**Reply by rtritto** [reply]
It works, thanks!

Maybe is better to write an example in doc or write another separate example

**Reply by rtritto** [reply]
It can be improved:
```diff
  } else {
+    document.title = getTitle(pageContext)    // OR document.title = pageContext.config.title
    setPageContext(reconcile(pageContext))
  }

-  const title = getTitle(pageContext)
-  if (title !== null) {
-    document.title = title
-  }
```

**Reply by brillout** [reply]
The plan is to have integrator packages, such as `vike-react` and `vike-solid`, to already provide a `title` implementation. The docs will be udpated with scuh "convention configs".

**Reply by brillout** [reply]
> It can be improved:

I see. I think you're right 👍.

**Reply by rtritto** [reply]
I would separate this use case in another example to have the simple example splitted

**Reply by brillout** [reply]
Good idea. I think that would actually be a good example of how to use `meta`. (FYI @harrytran998.)

Ticket: https://github.com/brillout/vite-plugin-ssr/issues/857.
